### PR TITLE
plotting: Don't reload all plot paths for each batch, do it once upfront

### PR DESCRIPTION
**Pull request by xdustinface**

### CODE REVIEW

File: chia/plotting/manager.py [hunk 1]
```diff
@@ -431,7 +431,7 @@ def plot_removed(test_path: Path):
                     del self.plot_filename_paths[filename]
 
             initial_value: Dict[Path, PlotInfo] = {}
-            self.plots = reduce(reduce_function, executor.map(process_file, all_filenames), initial_value)
+            self.plots = reduce(reduce_function, executor.map(process_file, plot_paths), initial_value)
```
Comment (arvidn) [inline chia/plotting/manager.py:434]: `all_filenames` was renamed to `plot_paths`, but they still contain the same data, right?

File: chia/plotting/manager.py [hunk 2]
```diff
@@ -407,7 +407,7 @@ def reduce_function(x: Dict, y: Dict) -> Dict:
 
             # First drop all plots we have in plot_filename_paths but not longer in the filesystem or set in config
             def plot_removed(test_path: Path):
-                return not test_path.exists() or test_path.parent not in plot_filenames
+                return not test_path.exists() or test_path.parent not in plot_directories
```
Comment (arvidn) [inline chia/plotting/manager.py:410]: this took a while to thread through the change as well, but `plot_directories` ends up being a `Set` of the same values as the previous keys in `plot_filenames`.

**Review by arvidn — APPROVED** [review]

File: chia/plotting/manager.py (hunk 1 above)
Comment (xdustinface) [reply chia/plotting/manager.py:434]: Correct

File: chia/plotting/manager.py (hunk 2 above)
Comment (xdustinface) [reply chia/plotting/manager.py:410]: Correct

**Pull request merged**